apps/communications/contacts/js/views/selector.js

```diff
@@ -0,0 +1,120 @@
+  var saveButton = selectorHeader.querySelector('#save-button');
+  var container = document.getElementById('selector-container');
+  var template = document.getElementById('contactToImportTemplate');
+
+  saveButton.addEventListener('click', importAll);
+  selectorHeader.addEventListener('action', cancelHandler);
+  container.addEventListener('click', clickHandler);
+
+  var contactsToImport = [];
+
+  function cancelHandler() {
+    selectorHeader.removeEventListener('action', cancelHandler);
+    saveButton.removeEventListener('click', importAll);
+    container.removeEventListener('click', clickHandler);
+    Contacts.cancel();
+  }
+
+  function clickHandler(event) {
+    // FIXME Seems to be a conflict when calling an activity from an activity.
+    var id = event.target.dataset.tempId;
+    ActivityHandler.mozContactParam = contactsToImport[id];
+    var activity = {
+      source: {
+        name: 'new',
+        data: {
+          type: ['text/vcard'],
+          allowSave: true,
+          params: {
+            mozContactParam: true
+          }
+        }
+      }
+    };
+    ActivityHandler.handle(activity);
+  }
+
+  function importAll() {
+    var numDupsMerged = 0,
+        importedContacts = 0;
+
+    contactsToImport.forEach((contact, index) => {
+      contacts.Matcher.match(contact, 'passive', {
+        onmatch: matches => {
+          var callbacks = {
+            success: function(mergedContact) {
+              numDupsMerged++;
+              _doContinue(index, true);
+            },
+            error: _doContinue.bind(this, index, true)
+          };
+          contacts.adaptAndMerge(contact, matches, callbacks);
+        },
+        onmismatch: function() {
+          var saving = navigator.mozContacts.save(contact);
+
+          saving.onsuccess = () => {
+            _doContinue(index);
+          };
+
+          saving.onerror = (err) => {
+            console.error(err);
+            _doContinue(index, true);
+          };
+        }
+      });
+    });
+
+    function _doContinue(index, failed) {
+      if (!failed) {
+        importedContacts++;
+      }
+
+      if (contactsToImport.length - 1 <= index) {
+        utils.status.show({
+          id: 'memoryCardContacts-imported3', // FIXME show better message.
+          args: {n: importedContacts}
+        }, {
+          id: 'contactsMerged',
+          args: {numDups: numDupsMerged}
+        });
+
+        window.setTimeout(cancelHandler, 1500); // Allows to see status message.
+      }
+    }
+  }
+
+  var addContact = contacts => {
+    if (!Array.isArray(contacts)) {
+      contacts = [contacts];
+    }
+    contacts.forEach( _doAddContact);
+
+    function _doAddContact(contact) {
+      var clone = document.importNode(template.content, true);
+
+      var contactData = clone.querySelectorAll('p');
+      contactData[0].textContent = contact.name[0];
+      contactData[1].textContent = contact.tel && contact.tel[0].value;
```

Comment:
no. you need to to check if contact.tel is an array and if the first element of the array is defined